### ISSUE TITLE
Add `ignoreFunctions: []` to `color-*`

### ISSUE THREAD

**Issue by jsmitty12** [stylelint/stylelint#9457]
### What is the problem you're trying to solve?

I have CSS rules that include code like this:

```css
button {
  color: var(--primary-text-color, #000); // flagged by color-no-hex
  background-color: var(--surface-0, white); // flagged by color-named
}
```

The comments indicate which rule flags each line with a violation. For my project, it's acceptable for the hex/named colors since they are fallbacks when retrieving the custom property.

### What solution would you like to see?

I would like to see an option for the rules that allow me to tell stylelint to ignore violations that are in the fallback position for a custom property retrieval.

**Comment by jeddy3** [comment]
@jsmitty12 Thanks for the request and for using the template.

Both proposals sound good.

Let's add a generic `ignoreFunctions: []` [secondary option](https://stylelint.io/contributor-guide/rules#user-defined-ignore) to both rules so that the options can be configured for other use cases.

For example:

```json
{
  "color-named": [true, { "ignoreFunctions": ["var", "/^--/"] } ]
}
```

The `color-named` rule already has an [`ignore: ["inside-function"]`](https://stylelint.io/user-guide/rules/color-named#inside-function) secondary option, and this new option will be the more targeted version of that.

I've labelled the issue as ready to implement. Please consider [contributing](https://stylelint.io/contributing) if you have time.

You can use the [`ignoreFunctions` option in `length-zero-no-unit`](https://stylelint.io/user-guide/rules/length-zero-no-unit/#ignorefunctions) as a blueprint.
